tests/scalers/aws-dynamodb-stream.test.ts

```diff
@@ -0,0 +1,189 @@
+import test from 'ava'
+import * as fs from 'fs'
+import * as sh from 'shelljs'
+import * as tmp from 'tmp'
+import { CreateTableCommand, DescribeTableCommand, DeleteTableCommand, DynamoDBClient,} from '@aws-sdk/client-dynamodb'
+import { DynamoDBStreamsClient, DescribeStreamCommand } from '@aws-sdk/client-dynamodb-streams'
+import { createNamespace, sleep, waitForDeploymentReplicaCount } from './helpers'
+
+const awsRegion = 'ap-northeast-1'
+const awsAccessKey = process.env['AWS_ACCESS_KEY'];
+const awsSecretKey =  process.env['AWS_SECRET_KEY'];
+const dynamoDBStreamNamespace = 'keda-test'
+const dynamoDBTableName = 'keda-table01'
+const nginxDeploymentName = 'nginx-deployment'
+
+let dynamoDBClient;
+let dynamoDBStreamsClient;
+let dynamoDBStreamShardNum;
+
+test.before(async t => {
+
+  // setup dynamodb client
+  dynamoDBClient = new DynamoDBClient({
+    region: awsRegion,
+    credentials: {
+      accessKeyId: awsAccessKey,
+      secretAccessKey: awsSecretKey
+    }
+  });
+
+  createNamespace(dynamoDBStreamNamespace)
+
+  // Create table
+  let params = {
+    TableName: dynamoDBTableName,
+    KeySchema: [
+      { AttributeName: 'id', KeyType: 'HASH' },  //Partition key
+    ],
+    AttributeDefinitions: [
+      { AttributeName: 'id', AttributeType: 'S' },
+    ],
+    BillingMode: 'PAY_PER_REQUEST',
+    StreamSpecification: {
+      StreamEnabled: true,
+      StreamViewType: 'NEW_IMAGE'
+    }
+  }
+  let createTableCmd = new CreateTableCommand(params)
+  await dynamoDBClient.send(createTableCmd)
+  console.log("table is created!!")
+
+  // Get streamArn for the created dynamodb table
+  let describeTableCommand = new DescribeTableCommand({TableName: dynamoDBTableName});
+  let dbResponse = await dynamoDBClient.send(describeTableCommand);
+  const latestStreamArn = ( dbResponse.Table !== undefined ) ? dbResponse.Table.LatestStreamArn : ""
+
+  await sleep(10000)
+
+  // Get Shard Num
+  dynamoDBStreamsClient = new DynamoDBStreamsClient({
+    region: awsRegion,
+    credentials: {
+      accessKeyId: awsAccessKey,
+      secretAccessKey: awsSecretKey
+    }
+  });
+  let describeStreamCommand = new DescribeStreamCommand({
+    StreamArn: latestStreamArn
+  })
+
+  let dbsResponse = await dynamoDBStreamsClient.send(describeStreamCommand)
+  const shards = (dbsResponse.StreamDescription !== undefined) ? dbsResponse.StreamDescription.Shards : undefined 
+  dynamoDBStreamShardNum = (( shards !== undefined ) ? shards.length : 0)
```

Comment:
shouldn't this fail if shard count is 0?

Reply:
As you say, this should fail if the count is 0.  
So I added the check like this https://github.com/kedacore/keda/pull/3121/commits/593eebbc4032c5de9529919c11e32e270a626b0e